modules/extension/mbstyle/src/main/java/org/geotools/mbstyle/expression/MBLookup.java

```diff
@@ -56,16 +56,22 @@ public Expression lookupAt() {
     public Expression lookupGet() {
         if (json.size() == 2 || json.size() == 3) {
```

Comment:
could you put some comments in the code here (maybe an example)?  it's hard to determine what all the json.size() == X mean.

Also, I think get requires one argument - ["get", "propName"]

Reply:
There is actually an option, documented in the javadoc above for three arguments.
```
[ "get", key, object]
```

This can be used to pull out a specific value not from the feature, but from the results of another expression.  I was really smart I may be able to turn this into an propertyname with xpath - but I am not going to do premature optimization.

For the library we chose the geoapi approach of format shifting the specifrication into javadocs. So you do not have to look up everything each time to see what the code is trying to implement.